google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1042,4 +1045,100 @@ public Acl blobToPublicRead(String bucketName, String blobName, long blobGenerat
+        .setRequesterPays(true)
+        .build();
+
+    // Update the bucket, throws StorageException on failure
+    Bucket bucket = storage.update(bucketInfo);
+
+    System.out.println("Requester pay status for " + bucketName +": " + bucket.requesterPays());
+    // [END enable_requester_pays]
+    return bucket;
+  }
+
+  /**
+   * Example of disabling Requester pays on a bucket.
+   */
+  public Bucket disableRequesterPays(String bucketName) {
+    // [START disable_requester_pays]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of the bucket to disable requester-paying for, e.g. "my-bucket"
+    // String bucketName = "my-bucket"
+    BucketInfo bucketInfo = BucketInfo.newBuilder(bucketName)
+        .setRequesterPays(false)
+        .build();
+
+    // Update the bucket, throws StorageException on failure
+    Bucket bucket = storage.update(bucketInfo);
+
+    System.out.println("Requester pays status for " + bucketName +": " + bucket.requesterPays());
+    // [END disable_requester_pays]
+    return bucket;
+  }
+
+  /**
+   * Example of retrieving Requester pays status on a bucket.
+   */
+  public Bucket getRequesterPaysStatus(String bucketName) throws StorageException {
+    // [START get_requester_pays_status]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of the bucket to retrieve requester-pays status, eg. "my-bucket"
+    // String bucketName = "my-bucket"
+    // Retrieve the bucket, throws StorageException on failure
+    Bucket bucket = storage.get(bucketName);
+
+    System.out.println("Requester pays status : " + bucket.requesterPays());
+    // [END get_requester_pays_status]
+    return bucket;
+  }
+
+  /**
+   * Example of downloading a file using Requester pay.
+   */
+  public void downloadFileUsingRequesterPays(String projectId, String bucketName,
+      String srcFilename, Path destFilePath) throws IOException {
+    // [START storage_download_file_requester_pays]
+    // Instantiate a Google Cloud Storage client
+
+    // The project ID to bill
+    // String projectId = "my-billable-project-id";
+
+    // The name of the bucket to access
+    // String bucketName = "my-bucket";
+
+    // The name of the remote file to download
+    // String srcFilename = "file.txt";
+
+    // The path to which the file should be downloaded
+    // String destFilePath = "/local/path/to/file.txt";
+
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    BlobSourceOption option = BlobSourceOption.userProject(projectId);
+
+    // read blob in a single pass
+    byte[] readBytes = storage.readAllBytes(bucketName, srcFilename, option);
+
+    // write out to file
+    PrintStream out = new PrintStream(new FileOutputStream(destFilePath.toFile()));
```

Comment:
Just a note.
We now have `Blob.downloadTo(Path)` method, which seems applicable here, but unfortunatelly it has to be updated to `Blob.downloadTo(Path, BlobSourceOption...)` to support this use case. It is a rather small change, and probably should be done together with the remaining changes for the RequeseterPays feature. 
After that is done, please consider using it in samples which show how to download a file.

Reply:
Thanks for raising this @vam-google. I'll file an issue for this support.